Lot of improvements in Geometries, Adapters and overall performance

BathoryPeter opened the pull request:
- Z and M support for Geometries and all Adapters (where applicable)
- EMPTY geometry support
- complete centroid calculation, now using de facto standard weighted method
- rewritten WKB and EWKB adapter, supporting bigEndian, Z and M coordinates and SRID
- new TWKB adapter
- new OSM adapter (not complete yet, but has a promising multipolygon converter)
- reduced memory usage
- improved performance in Geometries, eg. built-in geometry operations, array manipulations, constructors, etc.
- improved type detection
- improved and more consistent block-level PHPDoc documentation (not ready)
- more strict OGR standard complaint geometry representation and operations, preserving Postgis compatibility

teohhanhui commented:
@BathoryPeter Can you delete the unused tags and branches in your fork to speed up composer install? Thanks! :)

p/s: You have not enabled Issues on that repo.

BathoryPeter committed:
refactoring: moved /lib to /src, renamed folder adapters to Adapter, removed "class" from file names, moved geoPHP.inc to src/geoPHP.php

BathoryPeter committed:
- PSR-4 namespaces
- Z and M support for Geometries and all Adapters (where applicable)
- EMPTY geometry support
- completed centroid calculation, now using de facto standard weighted method
- rewritten WKB and EWKB adapter, supporting bigEndian, Z and M coordinates and SRID
- reduced memory usage
- improved performance in Geometries, eg. built-in geometry operations, array manipulations, constructors, etc.
- improved type detection
- improved and more consistent block-level PHPDoc documentation (not ready)
- more strict OGR standard complaint geometry representation and operations, preserving Postgis compatibility

BathoryPeter committed:
- new TWKB adapter
- new OSM adapter (not complete yet, but has a basic multipolygon converter)

BathoryPeter committed:
fixed some warning

BathoryPeter committed:
fixed travis.yml

BathoryPeter committed:
Added basic geometry metadata support

BathoryPeter committed:
Implemented a new buildGeometries() helper function. It doesn't modify geometry structures (like geometryReduce()) just builds the most appropriate geometry from the given array of geometries

BathoryPeter committed:
Implemented (meta)data support in GPX and GeoJSON adapters

BathoryPeter committed:
fixed GPX adapter

BathoryPeter committed:
workaround a HHVM bug

BathoryPeter committed:
fix for php 5.4

BathoryPeter committed:
Start using geometry type constants

BathoryPeter committed:
documentation; minor fixes; make GeoHash->adjacent() static

BathoryPeter committed:
Added two custom exception

BathoryPeter committed:
Bug fixes and minor improvements in geoPHP.php tools
- fixed a multi-geometry bug in buildGeometry()
- changed geoPHP::load() to use buildGeometry as a last step (instead of geometryReduce)
- added some more Earth constant

BathoryPeter committed:
Added a simple performanceTest and polished other tests to deal with new exceptions

BathoryPeter committed:
Refactored Geometries, and polished geometry methods
- Add abstract geometry types like Curve, Surface, MultiGeometry
- move common methods to abstract classes
- add static ::fromArray() method (only to Point and LineString yet)
- comment methods and classes
- use UnsupportedMethodException::geos() exceptions
- rewrite boundary() methods to follow the OGC standard
- implemented MultiPoint::isSimple()
- lot of minor bug fixes and improvements

BathoryPeter committed:
added a new length calculation method to LineString using Vincenty formula
author: Marcus Jaschen, https://github.com/mjaschen/phpgeo

BathoryPeter committed:
New unit tests for geometries
- currently Point and LineString tests are complete (excluding geos fallbacks)
- added a phpunit.xml config, creating coverage report
- exclude old unit tests from the default config (they are obsolete, not real unit tests, should be rewritten)

BathoryPeter committed:
fixed LineStringTest on HHVM
(set tolerance in "equals" tests and skip tests where HHVM float precision isn't enough)

BathoryPeter committed:
Unit tests of MultiPoint

DaxServer commented on .gitignore:
```diff
@@ -1 +1,7 @@
 /vendor
+/tests/coverage/
+composer.lock
```
Probably include the lock file so that everyone forking and developing would have the same versions of package dependencies?

Commit:
Speed up parsing GPX files
Only converting malformed xml tags if parsing according to the schema don't bring result

Commit:
Update composer dev requirement

Commit:
Basic metadata support for KML reader

Commit:
fix buildGeometries() crash if input array contains only GeometryCollections

BathoryPeter committed:
remove duplicated line in KML adapter, it added each geometry twice :S
more descriptive exception messages

BathoryPeter committed:
drop PHP 5.4 support; update composer.json and travis.yml;
replace references to phayes repository in readme

BathoryPeter committed:
update composer.json to use proper license names

BathoryPeter committed:
add keywords to composer.json

Commit:
Unifies parameters of elevationGain() and elevationLoss()

Commit:
geoJSON reader: accept empty GeometryCollection and throw excepion only if it is malformed

Commit:
fix buildGeometry() to prevent losing Data:
do not create MultiGeometry from geometries if any of them has Data

Commit:
fix NAN result of haversineLength()

Commit:
Fix logic how isEmpty() should be determined for Collections

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/phayes/geoPHP/685562416ec6d22b9b3927e02ca0ddacf84ca646/phayes/geoPHP/e09098ee9ebb24e94c92ac4d85b2a74a1ea88607/_/%2F/code/?utm_source=gh_review) your code in 0.001 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/phayes/geoPHP/685562416ec6d22b9b3927e02ca0ddacf84ca646/phayes/geoPHP/e09098ee9ebb24e94c92ac4d85b2a74a1ea88607/_/%2F/code/?utm_source=gh_review) 
---

Commit:
GPX adapter: do not concatenate points of different trkseg elements

Commit:
GPX adapter: avoid creating invalid Linestring

Commit:
Updated Travis config: added 7.3, 7.4, removed 5.5, HHVM
Composer dropped support of HHVM
Travis build fails for PHP 5.5 because archives are only available on trusty

BathoryPeter commented:
Closing this, because my fork got too diverged in the past years.
If anybody is interested in a maintained fork, please take a look at https://github.com/funiq/geoPHP
